extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -1229,6 +1230,12 @@ public static Map<String, JpaPersistenceUnitModel> getModelPerPersistenceUnit(Hi
         Set<String> modelClassesWithPersistenceUnitAnnotations = new TreeSet<>();
 
         for (String modelClassName : jpaModel.getAllModelClassNames()) {
+
+            if(!jpaModel.getEntityClassNames().contains(modelClassName)) {
+                // We only care about class name here
+                continue;
+            }
+
```

Comment:
I don't think this is correct, as you can see within this for loop we add the `modelClassName` to the `model.allModelClassAndPackageNames()` on line 1257 even if it's not an entity.